### ISSUE TITLE
Using object-fit

### ISSUE THREAD

**Issue by Jmuccigr** [hakimel/reveal.js#1669]
I'm trying to use reveal.js for lectures where I like to have the image, proportioned correctly, of course, fill the window. My original thought was to use javascript to inject css to control the image relative width and height depending on the image's size and proportions, but then I stumbled upon `object-fit: contain` and after some fiddling have it working with Safari. That works for me, but I like to share these lectures with students, so it would be nice if it worked with FireFox and Chrome/ium, but it doesn't.

I use a custom css file, a modified version of the standard gray.css. [I attach it](https://github.com/hakimel/reveal.js/files/428478/grayfull.css.txt), but the key bits are here:

```javascript
/* Get rid of the inherited padding*/

.reveal .slides > section,
.reveal .slides > section > section {
  padding: 0;
}

/* Get slides to fill the whole window */

.reveal .slides {
  width: 100%;
  height: 100%;
  top: 0;
  margin-top: 0;
}

.reveal .slides > section figure img {
  margin: 0;
  border: none;
  box-shadow: none;
  padding: 0;
  object-fit: contain;
  min-height: 100%;
  max-width: 100%;
}

/* Hide the captions in this context*/

.reveal figcaption {
  display: none;
}
```

Without the min-height and max-width settings, the images don't expand to fill the window, but are left in their original sizes. I suspect this is because the image thinks its enclosing element is only as big as it is, perhaps due to the way reveal.js works. These attributes would then force the image to fill the screen and object-fit keeps it proportioned correctly.

Whether that's it or not, FireFox and Chromium both act as Safari does without those two attributes, that is, they show the images at their native sizes.

Any help would be appreciated. It seems to me that object-fit is a nice way to get this working without javascript, but I would take a solution that uses javascript as well, or something else that works.

**Comment by Jmuccigr** [comment]
No help on this?

**Comment by nimonian** [comment]
Isn't there a data="strech" attribute you can put in the image div?

**Comment by Jmuccigr** [comment]
Yeah, I suppose my question has to do with whether Safari is doing this right (and the others wrong) or vice versa, and whether there's another way to do this.
